x/oracle/abci.go

```diff
@@ -87,12 +87,17 @@ func EndBlocker(ctx sdk.Context, k keeper.Keeper) {
 		//---------------------------
 		// Do miss counting & slashing
 		voteTargetsLen := len(voteTargets)
-		for _, claim := range validatorClaimMap {
+		for key, claim := range validatorClaimMap {
 			// Skip abstain & valid voters
 			if int(claim.WinCount) == voteTargetsLen {
 				continue
 			}
 
+			// If a validator miss any single ballot,
+			// the one can't receive any reward for this voting period.
+			claim.Weight = 0
+			validatorClaimMap[key] = claim
```

Comment:
Is there any reason not to modify map directly? (i.e. validatorClaimMap[key].Weight = 0)

Reply:
yea in golang, seem that operation is not allowed